apps/bfd-data-fda/src/test/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookupTest.java

```diff
@@ -40,46 +48,57 @@ public class FdaDrugCodeDisplayLookupTest {
   /** fdaDrugCodeDisplays to be used during testing. */
   public FdaDrugCodeDisplayLookup fdaDrugCodeDisplay;
 
+  /** npiDataStream represents npiDataStream of the main FDA drug org file. */
+  InputStream npiDataStream =
+      Thread.currentThread().getContextClassLoader().getResourceAsStream(App.FDA_PRODUCTS_RESOURCE);
+
+  /** ndcProductHashMap represents a map of PRODUCTNDC and SUBSTANCENAME for test. */
+  Map<String, String> ndcProductHashMap = new HashMap<>();
+
   /** Clears fdaDrugCodeDisplay before each test. */
   @BeforeEach
   void setup() {
     fdaDrugCodeDisplay = null;
+    ndcProductHashMap.put(FAKE_DRUG_CODE, FAKE_DRUG_CODE_DISPLAY);
   }
 
-  /** Return Fake Drug Code when parameter is true. */
+  /** Return Fake Drug Code. */
   @Test
-  public void shouldReturnFakeDrugCodeWhenConstructorSetToTrue() {
-    fdaDrugCodeDisplay = FdaDrugCodeDisplayLookup.createDrugCodeLookupForTesting();
+  public void shouldReturnFakeDrugCode() throws IOException {
+
+    fdaDrugCodeDisplay = new FdaDrugCodeDisplayLookup(ndcProductHashMap);
+
     String drugCodeDisplay =
         fdaDrugCodeDisplay.retrieveFDADrugCodeDisplay(Optional.of(FAKE_DRUG_CODE_NUMBER));
     assertNotEquals(null, drugCodeDisplay);
   }
 
-  /** Do Not Return Fake Drug Code when parameter is false. */
+  /** Doesn't return Fake Drug Code. */
   @Test
-  public void shouldNotReturnFakeDrugCodeWhenConstructorSetToFalse() {
-    fdaDrugCodeDisplay = FdaDrugCodeDisplayLookup.createDrugCodeLookupForProduction();
+  public void shouldNotReturnFakeDrugCode() throws IOException {
+    fdaDrugCodeDisplay = new FdaDrugCodeDisplayLookup(npiDataStream);
     String drugCodeDisplay =
         fdaDrugCodeDisplay.retrieveFDADrugCodeDisplay(Optional.of(FAKE_DRUG_CODE_NUMBER));
     assertEquals(null, drugCodeDisplay);
   }
 
-  /** Return Fake Drug Code Display when parameter is true. */
+  /** Return Fake Drug Code Display. */
   @Test
-  public void shouldReturnFakeDrugCodeDisplayWhenConstructorSetToTrue() {
-    fdaDrugCodeDisplay = FdaDrugCodeDisplayLookup.createDrugCodeLookupForTesting();
+  public void shouldReturnFakeDrugCodeDisplay() {
+    fdaDrugCodeDisplay = new FdaDrugCodeDisplayLookup(ndcProductHashMap);
+
     String drugCodeDisplay =
         fdaDrugCodeDisplay.retrieveFDADrugCodeDisplay(Optional.of(FAKE_DRUG_CODE_NUMBER));
-    assertEquals(FdaDrugCodeDisplayLookup.FAKE_DRUG_CODE_DISPLAY, drugCodeDisplay);
+    assertEquals(FAKE_DRUG_CODE_DISPLAY, drugCodeDisplay);
   }
 
-  /** Do not Return Fake Drug Code Display when parameter is false. */
+  /** Do not Return Fake Drug Code Display. */
   @Test
-  public void shouldNotReturnFakeDrugCodeDisplayWhenConstructorSetToFalse() {
-    fdaDrugCodeDisplay = FdaDrugCodeDisplayLookup.createDrugCodeLookupForProduction();
+  public void shouldNotReturnFakeDrugCodeDisplayWhenConstructorSetToFalse() throws IOException {
```

Comment:
Mind taking a look at some of these function names? A lot of these don't make sense anymore since we're not setting anything to false.